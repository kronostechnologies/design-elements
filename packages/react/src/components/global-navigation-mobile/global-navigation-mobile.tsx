import React, { ReactElement, ReactNode } from 'react';
import styled from 'styled-components';

const Container = styled.div`
  background-color: ${props => props.theme.greys.white};
  font-size: 1.125rem;
  height: 100vh;
  left: 0%;
  margin-top: 2.75rem;
  overflow-y: auto;
  padding: 0 30px;
  position: fixed;
  top: 0%;
  transform: translate(${(props: {open: boolean}) => props.open ? '0%' : '100%'});
  transition: all 300ms;
  width: 100%;
  z-index: 100;

  > .side-panel {
    margin-top: 0;
  }

  li {
    margin-left: 24px;
    padding: 12px 0;

    svg {
      margin-left: -24px;
    }

    .external {
      color: ${props => props.theme.greys['dark-grey']};

      &:hover {
        color: ${props => props.theme.greys.black};
        text-decoration: none;
      }

      &:visited {
        color: ${props => props.theme.greys['dark-grey']};

        &:hover {
          color: ${props => props.theme.greys.black};
        }
      }
    }

    &:first-child {
      padding-top: 0;
    }

    &:last-child {
      padding-bottom: 0;
    }
  }

  ul {
    list-style: none;
    margin: 0;
    padding: 24px 0;
  }
`;

interface GlobalNavMobileProps {
    children?: ReactNode;
    open: boolean;
}

export function GlobalNavigationMobile({ children, open }: GlobalNavMobileProps): ReactElement {
    return (
        <>
            <Container open={open} className="side-panel">
                {children}
            </Container>
        </>
    );
}
