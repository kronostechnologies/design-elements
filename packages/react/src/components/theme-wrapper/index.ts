export { ThemeWrapper } from './theme-wrapper';
export type { ThemeWrapperProps } from './theme-wrapper';
