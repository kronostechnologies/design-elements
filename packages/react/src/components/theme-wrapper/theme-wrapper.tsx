import React, { ReactElement } from 'react';
import { ThemeProvider } from 'styled-components';

import {Â equisoftTheme } from '../../themes/equisoft';

export interface Theme {
    main: {
        'primary-1.1': string,
        'primary-1.2': string,
        'primary-1.3': string,
        'primary-2': string,
        'primary-3': string,
        'secondary-4.1': string,
        'secondary-4.2': string,
        'secondary-4.3': string,
    };
    greys: {
        'white': string,
        'colored-white': string,
        'light-grey': string,
        'grey': string,
        'mid-grey': string,
        'dark-grey': string,
        'black': string,
    };
    notifications: {
        'success-1.1': string,
        'success-1.2': string,
        'success-1.3': string,
        'error-2.1': string,
        'error-2.2': string,
        'alert-3.1': string,
        'alert-3.2': string,
        'alert-3.3': string,
    };
}

interface ThemeWrapperProps {
    children: ReactElement;
    theme?: Theme;
}

export function ThemeWrapper({ children, theme }: ThemeWrapperProps): ReactElement {
    let selectedTheme = theme;
    if (selectedTheme) {
        if (Object.entries(selectedTheme).length === 0 && selectedTheme.constructor === Object) {
            selectedTheme = equisoftTheme;
        }
    } else {
        selectedTheme = equisoftTheme;
    }
    return (
        <ThemeProvider theme={selectedTheme}>
            {children}
        </ThemeProvider>
    );
}
