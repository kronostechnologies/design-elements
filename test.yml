npmRegistries: 
    //registry.npmjs.org:
    npmAlwaysAuth: true
    npmAuthToken: 
