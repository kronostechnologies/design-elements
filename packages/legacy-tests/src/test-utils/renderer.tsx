import { CommonWrapper, mount, MountRendererProps, ReactWrapper, render } from 'enzyme';
import { act, Component, FunctionComponent, PropsWithChildren, ReactElement } from 'react';
import { MemoryRouter } from 'react-router-dom';
import { DesignSystem, DesignSystemProps } from '~/components/design-system';
import { ThemeWrapper } from '~/components/theme-wrapper/theme-wrapper';

export const AllProviders: FunctionComponent<PropsWithChildren<DesignSystemProps>> = ({
    children,
    language,
    theme,
    staticDevice,
}) => (
    <MemoryRouter future={{ v7_startTransition: false, v7_relativeSplatPath: false }}>
        <DesignSystem language={language} staticDevice={staticDevice} theme={theme}>
            {children}
        </DesignSystem>
    </MemoryRouter>
);

interface WrappingComponentProps {
    wrappingComponentProps?: DesignSystemProps;
}

type Options = MountRendererProps & WrappingComponentProps;

export function mountWithProviders<C extends Component, P = C['props'], S = C['state']>(
    component: ReactElement<P>,
    options: Options = {},
): ReactWrapper<P, S, C> {
    return mount(component, {
        ...options,
        wrappingComponent: AllProviders as Options['wrappingComponent'],
    });
}

export function mountWithTheme<C extends Component, P = C['props'], S = C['state']>(
    component: ReactElement<P>,
    options?: MountRendererProps,
): ReactWrapper<P, S, C> {
    return mount(component, {
        wrappingComponent: ThemeWrapper as MountRendererProps['wrappingComponent'],
        ...options,
    });
}

export function renderWithTheme(
    component: ReactElement,
): cheerio.Cheerio {
    return render(
        <ThemeWrapper>
            {component}
        </ThemeWrapper>,
    );
}

export async function actUpdate<C extends Component, P = C['props'], S = C['state']>(
    wrapper: CommonWrapper<P, S, C>,
): Promise<void> {
    await act(async () => {
        wrapper.update();
    });
}

export async function actAndWaitForEffects<C extends Component, P = C['props'], S = C['state']>(
    wrapper: ReactWrapper<P, S, C>,
    action: () => void,
): Promise<ReactWrapper<P, S, C>> {
    await act(async () => action());
    await actUpdate(wrapper);
    return wrapper;
}
