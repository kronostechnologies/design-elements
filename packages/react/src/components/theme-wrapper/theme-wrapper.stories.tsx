import React from 'react';

import { testTheme } from '@design-elements/themes/test-theme';
import { Button } from '../buttons/button';
import { ThemeWrapper } from './theme-wrapper';

export default {
    title: 'Theme Wrapper',
    component: ThemeWrapper,
};

export const normal = () => (
    <>
        <ThemeWrapper theme={testTheme}>
            <div>
                <h3 style={{ marginTop: '0' }} >With wrapper (test theme)</h3>
                <Button label="Primary" buttonType="primary" disabled={false} />
                <Button label="Secondary" buttonType="secondary" disabled={false} />
                <Button label="Tertiary" buttonType="tertiary" disabled={false} />
            </div>
        </ThemeWrapper>
        <div>
            <h3>Without wrapper (default Equisoft theme)</h3>
            <Button label="Primary" buttonType="primary" disabled={false} />
            <Button label="Secondary" buttonType="secondary" disabled={false} />
            <Button label="Tertiary" buttonType="tertiary" disabled={false} />
        </div>
    </>
);
