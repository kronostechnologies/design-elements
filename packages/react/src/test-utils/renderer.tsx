import { ThemeWrapped } from '@design-elements/test-utils/theme-wrapped';
import { mount, ReactWrapper, render } from 'enzyme';
import React, { Component, ReactElement } from 'react';
import { MemoryRouter } from 'react-router-dom';
import { DeviceContextProvider } from '../components/device-context-provider/device-context-provider';

export function mountWithProviders<C extends Component, P = C['props'], S = C['state']>(
    component: ReactElement<P>,
): ReactWrapper<P, S, C> {
    return mount(component, {
        wrappingComponent: AllProviders,
    });
}

export function renderWithProviders(
    component: ReactElement,
): Cheerio {
    return render(<AllProviders>{component}</AllProviders>);
}

function AllProviders({ children }: { children: ReactElement }): ReactElement {
    return (
        <MemoryRouter>
            <DeviceContextProvider>
                {ThemeWrapped(children)}
            </DeviceContextProvider>
        </MemoryRouter>
    );
}
