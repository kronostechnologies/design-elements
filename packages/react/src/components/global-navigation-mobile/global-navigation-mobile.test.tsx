import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeWrapped } from '../theme-wrapper/theme-wrapper.test';
import { GlobalNavigationMobile } from './global-navigation-mobile';

describe('Global Navigation Mobile', () => {
    test('is open', () => {
        const tree = renderer.create(
            ThemeWrapped(
                <GlobalNavigationMobile open>
                    <ul>
                        <li>Test</li>
                    </ul>
                </GlobalNavigationMobile>,
            ),
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('is closed on right side', () => {
        const tree = renderer.create(
            ThemeWrapped(
                <GlobalNavigationMobile open={false}>
                    <ul>
                        <li>Test</li>
                    </ul>
                </GlobalNavigationMobile>,
            ),
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('is closed on left side', () => {
        const tree = renderer.create(
            ThemeWrapped(
                <GlobalNavigationMobile open={false} menuOrigin="left">
                    <ul>
                        <li>Test</li>
                    </ul>
                </GlobalNavigationMobile>,
            ),
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
