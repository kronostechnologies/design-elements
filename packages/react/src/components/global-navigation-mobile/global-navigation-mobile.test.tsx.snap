// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global Navigation Mobile is closed on left side 1`] = `
.c0 {
  background-color: #FFFFFF;
  font-size: 1.125rem;
  height: 100vh;
  left: 0%;
  margin-top: 2.75rem;
  overflow-y: auto;
  padding: 0 30px;
  position: fixed;
  top: 0%;
  -webkit-transform: translate(-100%);
  -ms-transform: translate(-100%);
  transform: translate(-100%);
  -webkit-transition: all 300ms;
  transition: all 300ms;
  width: 100%;
  z-index: 100;
}

.c0 > .side-panel {
  margin-top: 0;
}

.c0 li {
  margin-left: 24px;
  padding: 12px 0;
}

.c0 li svg {
  margin-left: -24px;
}

.c0 li .external {
  color: #57666E;
}

.c0 li .external:hover {
  color: #000000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 li .external:visited {
  color: #57666E;
}

.c0 li .external:visited:hover {
  color: #000000;
}

.c0 li:first-child {
  padding-top: 0;
}

.c0 li:last-child {
  padding-bottom: 0;
}

.c0 ul {
  list-style: none;
  margin: 0;
  padding: 24px 0;
}

<div
  className="c0 side-panel"
  open={false}
  origin="left"
>
  <ul>
    <li>
      Test
    </li>
  </ul>
</div>
`;

exports[`Global Navigation Mobile is closed on right side 1`] = `
.c0 {
  background-color: #FFFFFF;
  font-size: 1.125rem;
  height: 100vh;
  left: 0%;
  margin-top: 2.75rem;
  overflow-y: auto;
  padding: 0 30px;
  position: fixed;
  top: 0%;
  -webkit-transform: translate(100%);
  -ms-transform: translate(100%);
  transform: translate(100%);
  -webkit-transition: all 300ms;
  transition: all 300ms;
  width: 100%;
  z-index: 100;
}

.c0 > .side-panel {
  margin-top: 0;
}

.c0 li {
  margin-left: 24px;
  padding: 12px 0;
}

.c0 li svg {
  margin-left: -24px;
}

.c0 li .external {
  color: #57666E;
}

.c0 li .external:hover {
  color: #000000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 li .external:visited {
  color: #57666E;
}

.c0 li .external:visited:hover {
  color: #000000;
}

.c0 li:first-child {
  padding-top: 0;
}

.c0 li:last-child {
  padding-bottom: 0;
}

.c0 ul {
  list-style: none;
  margin: 0;
  padding: 24px 0;
}

<div
  className="c0 side-panel"
  open={false}
  origin="right"
>
  <ul>
    <li>
      Test
    </li>
  </ul>
</div>
`;

exports[`Global Navigation Mobile is open 1`] = `
.c0 {
  background-color: #FFFFFF;
  font-size: 1.125rem;
  height: 100vh;
  left: 0%;
  margin-top: 2.75rem;
  overflow-y: auto;
  padding: 0 30px;
  position: fixed;
  top: 0%;
  -webkit-transform: translate(0%);
  -ms-transform: translate(0%);
  transform: translate(0%);
  -webkit-transition: all 300ms;
  transition: all 300ms;
  width: 100%;
  z-index: 100;
}

.c0 > .side-panel {
  margin-top: 0;
}

.c0 li {
  margin-left: 24px;
  padding: 12px 0;
}

.c0 li svg {
  margin-left: -24px;
}

.c0 li .external {
  color: #57666E;
}

.c0 li .external:hover {
  color: #000000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 li .external:visited {
  color: #57666E;
}

.c0 li .external:visited:hover {
  color: #000000;
}

.c0 li:first-child {
  padding-top: 0;
}

.c0 li:last-child {
  padding-bottom: 0;
}

.c0 ul {
  list-style: none;
  margin: 0;
  padding: 24px 0;
}

<div
  className="c0 side-panel"
  open={true}
  origin="right"
>
  <ul>
    <li>
      Test
    </li>
  </ul>
</div>
`;

exports[`Theme Wrapper Returns component with default theme 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: inherit;
  border: 1px solid;
  border-radius: 1.25rem;
  box-sizing: border-box;
  color: inherit;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 0.75rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.025rem;
  -moz-letter-spacing: 0.025rem;
  -ms-letter-spacing: 0.025rem;
  letter-spacing: 0.025rem;
  line-height: normal;
  min-height: 2rem;
  min-width: 2rem;
  outline: none;
  padding: 0.5rem 1rem;
  text-transform: uppercase;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: #EF483E;
  border-color: #EF483E;
  color: #FFFFFF;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c0 > svg {
  color: inherit;
}

.c0:hover {
  background-color: #D41F14;
  border-color: #D41F14;
}

.c0:disabled,
.c0:disabled:focus,
.c0:disabled:hover {
  background-color: #FF847C;
  border-color: #FF847C;
}

<button
  className="c0"
  onClick={[Function]}
>
  
</button>
`;

exports[`Theme Wrapper Returns component with test theme 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: inherit;
  border: 1px solid;
  border-radius: 1.25rem;
  box-sizing: border-box;
  color: inherit;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 0.75rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.025rem;
  -moz-letter-spacing: 0.025rem;
  -ms-letter-spacing: 0.025rem;
  letter-spacing: 0.025rem;
  line-height: normal;
  min-height: 2rem;
  min-width: 2rem;
  outline: none;
  padding: 0.5rem 1rem;
  text-transform: uppercase;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: #B3974B;
  border-color: #B3974B;
  color: #FFFFFF;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c0 > svg {
  color: inherit;
}

.c0:hover {
  background-color: #FFD86B;
  border-color: #FFD86B;
}

.c0:disabled,
.c0:disabled:focus,
.c0:disabled:hover {
  background-color: #736130;
  border-color: #736130;
}

<button
  className="c0"
  onClick={[Function]}
>
  
</button>
`;
